client/src/cordova/android/OutlineAndroidLib/outline/src/main/aidl/org/outline/VpnServiceError.aidl

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.outline;
+
+/*
+ * This class represents a structured error from the Android VPN service.
+ * The code field is a string that can be used to determine the error category in Java.
+ * The errorJson contains a JSON string of the error details that can be passed back to TypeScript.
+ */
+parcelable VpnServiceError {
```

Comment:
There's nothing specific to VPNs here. Perhaps call it `JsonCodeError`, `JsonDetailsError` or something like that?

Reply:
Renamed to `DetailedJsonError`.